identity/v3userpass.go

```diff
@@ -186,7 +189,7 @@ func v3KeystoneAuth(c *goosehttp.Client, v interface{}, url string) (*AuthDetail
 	}
 	tok := req.RespHeaders.Get("X-Subject-Token")
 	if tok == "" {
-		return nil, fmt.Errorf("authentication failed")
+		return nil, gooseerrors.NewUnauthorisedf(nil, "", "token value associated with X-Subject-Token key is empty")
```

Comment:
No need for ""
Plus too much internal detail. Maybe just "empty auth token received"